### PR TITLE
Fcntl 0xc0000007, 0xc0000015, 0xc000001c

### PR DESCRIPTION
**Pull request by clienthax**
Needed for S357 games.
Needs to be hooked up to a config.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 1]
```diff
@@ -1649,17 +1649,54 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
 
 	case 0xc0000007: // cellFsArcadeHddSerialNumber
 	{
-		break;
+		const auto arg = vm::static_ptr_cast<lv2_file_c000007>(_arg);
+
+		// No idea what these are meant to be
+		std::string unk1 = "";
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1655]: The e = "" is redundant

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1655]: I think it's about the meaning it tries to convey - that the correct strings are unknown.

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1655]: conatexpr std::string_view should be used here btw.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 2]
```diff
@@ -1649,17 +1649,54 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
...
+		memcpy(arg->unk1.get_ptr(), unk1.c_str(), unk1.size());
+
+		std::string unk2 = "";
+		memcpy(arg->unk2.get_ptr(), unk2.c_str(), unk2.size());
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/lv2/sys_fs.cpp:1659]: Same here.

File: rpcs3/Emu/Cell/lv2/sys_fs.cpp [hunk 3]
```diff
@@ -1649,17 +1649,54 @@ error_code sys_fs_fcntl(ppu_thread& ppu, u32 fd, u32 op, vm::ptr<void> _arg, u32
 
 	case 0xc0000007: // cellFsArcadeHddSerialNumber
 	{
-		break;
+		const auto arg = vm::static_ptr_cast<lv2_file_c000007>(_arg);
+
+		// No idea what these are meant to be
+		std::string unk1 = "";
+		memcpy(arg->unk1.get_ptr(), unk1.c_str(), unk1.size());
```
[the remark this answers is not in the document]
Comment (clienthax) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1656]: That's not really an issue currently as the buffer is blank either way

[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_fs.cpp:1656]: You need to write null charcter at tge end, do size() + 1 and it will take care of that.

**Pull request merged**